pkg/guardian/redis_conf_store.go

```diff
@@ -425,12 +639,23 @@ func (rs *RedisConfStore) RemoveRouteRateLimits(urls []url.URL) error {
 func (rs *RedisConfStore) GetRouteRateLimit(url url.URL) Limit {
 	rs.conf.RLock()
 	defer rs.conf.RUnlock()
-	limit, _ := rs.conf.routeRateLimits[url]
-	return limit
+	if rs.conf.useDeprecatedRouteRateLimit[url] {
+		return rs.conf.routeRateLimits[url]
+	}
+	return rs.conf.rateLimitsByURL[url].Spec.Limit
```

Comment:
Okay, I fixed GetRouteRateLimit and GetJail to check the existence of the map key and return an empty Limit/Jail if it's not found, matching the [previous behavior](https://github.com/dollarshaveclub/guardian/blob/37cbcedc043d3db820986737af2b215301133e83/pkg/guardian/redis_conf_store.go#L424). These empty objects will implicitly have Enabled set to False, so no limit will be applied to a URL that doesn't have a RateLimit or Jail configured for it. Spec can't be nil as it's a regular value field in the RateLimitConfig struct.

It definitely makes me a bit nervous that these methods can't return an error, or a least a pointer to indicate the absence of a rule for a certain URL. This is a consequence of this method needing to fulfill the [guardian.RouteRateLimitStore](https://github.com/dollarshaveclub/guardian/blob/5d315f967b97d487ec0cffcfe2defb5b3366fe8a/pkg/guardian/routeratelimit.go#L14) interface, which is how the rest of Guardian interacts with RedisConfStore to know what should be limited (similar to [JailProvider](https://github.com/dollarshaveclub/guardian/blob/5d315f967b97d487ec0cffcfe2defb5b3366fe8a/pkg/guardian/jail.go#L53), [GlobalLimitProvider](https://github.com/dollarshaveclub/guardian/blob/d7019f5db179f4dc84aff932cf09ef37ebaedd48/pkg/guardian/ipratelimit.go#L14), and [ReportOnlyProvider](https://github.com/dollarshaveclub/guardian/blob/d7019f5db179f4dc84aff932cf09ef37ebaedd48/pkg/guardian/server.go#L12)). Instead of this implicitly-disabled behavior, I think it would be good to change these interfaces to more clearly signal what is going on.